hikari/guilds.py

```diff
@@ -1302,6 +1302,209 @@ class WelcomeChannel:
     """ID of the emoji shown in the welcome screen channel if it's set to a custom emoji."""
 
 
+@typing.final
+class OnboardingPromptType(int, enums.Enum):
+    """The type of onboarding prompt."""
+
+    MULTIPLE_CHOICE = 0
+    """A multiple choice prompt."""
+
+    DROPDOWN = 1
+    """A dropdown prompt."""
+
+
+@typing.final
+class OnboardingMode(int, enums.Enum):
+    """The mode of onboarding."""
+
+    ONBOARDING_DEFAULT = 0
+    """Counts only Default Channels towards constraints."""
+
+    ONBOARDING_ADVANCED = 1
+    """Counts Default Channels and Questions towards constraints."""
+
+
+@attrs_extensions.with_copy
+@attrs.define(kw_only=True, weakref_slot=False)
+class GuildOnboarding:
+    """Used to represent guild onboarding settings on Discord."""
+
+    app: traits.RESTAware = attrs.field(
+        repr=False, eq=False, hash=False, metadata={attrs_extensions.SKIP_DEEP_COPY: True}
+    )
+    """Client application that models may use for procedures."""
+
+    guild_id: snowflakes.Snowflake = attrs.field(repr=True)
+    """ID of the guild this onboarding is part of."""
+
+    prompts: typing.Sequence[OnboardingPrompt] = attrs.field(repr=False)
+    """Prompts shown during onboarding and in customize community."""
+
+    default_channel_ids: typing.Sequence[snowflakes.Snowflake] = attrs.field(repr=False)
+    """Channel IDs that members get opted into automatically."""
+
+    enabled: bool = attrs.field(repr=True)
+    """Whether the guild onboarding is enabled."""
+
+    mode: OnboardingMode = attrs.field(repr=True)
+    """The mode of onboarding."""
+
+    def get_guild(self) -> typing.Optional[Guild]:
```

Comment:
@davfsa could you take a look at these get and fetch functions i added? i think its nice to have or do you think its just bloat?

Reply:
ok 👍🏼